test/integration/components/index.spec.ts

```diff
@@ -0,0 +1,24 @@
+import * as path from 'path';
+import { ToolpadHome } from '../../models/ToolpadHome';
+import { test } from '../../playwright/test';
+import { readJsonFile } from '../../utils/fs';
+
+test('components', async ({ page }) => {
+  const dom = await readJsonFile(path.resolve(__dirname, './componentsDom.json'));
+  const { id: pageId } = Object.values(dom.nodes).find(
+    (node: any) => node.name === 'components',
+  ) as any;
+
+  const homeModel = new ToolpadHome(page);
+  await homeModel.goto();
+  const app = await homeModel.createApplication({ dom });
+
+  await page.goto(`/app/${app.id}/preview/pages/${pageId}`);
+
+  await page.locator('text="foo button"').waitFor({ state: 'visible' });
```

Comment:
this works but maybe it should be more easy to understand which component each check is testing? maybe comments will do?

Reply:
nevermind i see the labels, didn't see it immediately though